Sources/PackageDescription/Version.swift

```diff
@@ -25,13 +25,17 @@ public struct Version {
     }
 
     public init?(_ characters: String.CharacterView) {
-        let components = characters.split(".", maxSplit: 2, allowEmptySlices: true).map(String.init).flatMap{ Int($0) }.filter{ $0 >= 0 }
+        let components = Version.componentsFromCharacters(characters)
         guard components.count == 3 else { return nil }
 
         self.major = components[0]
         self.minor = components[1]
         self.patch = components[2]
     }
+    
+    private static func componentsFromCharacters(characters: String.CharacterView) -> [Int] {
+        return characters.split(".", maxSplit: 2, allowEmptySlices: true).flatMap{ Int(String($0)) }.filter{ $0 >= 0 }
```

Comment:
I agree. But both versions of the old code (lines 28 and 168) as well as the new code return `[1, 0, 0]` for that string. My comments were made under the assumption that the purpose of this commit was to reduce code duplication while preserving existing functionality.

If you think this commit should also add functionality to handle invalid strings like that, I'd suggest putting it into an array of optionals: `let components = characters.split(".", maxSplit: 2, allowEmptySlices: true).map{ Int(String($0)) }`, or even put those values into a 3-tuple.